app/src/main/java/com/github/kulebin/myfoursquareapp/MainActivity.java

```diff
@@ -36,24 +58,75 @@ public void onItemSelected(final String pVenueId) {
         if (isTwoPaneMode) {
             findViewById(R.id.text_no_venue_selected).setVisibility(View.GONE);
             getSupportFragmentManager().beginTransaction()
-                    .replace(R.id.container_venue_detail, VenueDetailFragment.newInstance(pVenueId, true), VenueDetailFragment.TAG)
+                    .replace(R.id.container_venue_detail, VenueDetailFragment.newInstance(pVenueId), VenueDetailFragment.TAG)
                     .commit();
         } else {
             getSupportFragmentManager().beginTransaction()
                     .replace(R.id.container, VenueDetailFragment.newInstance(pVenueId), VenueDetailFragment.TAG)
                     .addToBackStack(null)
                     .commit();
+
+            isDrawerEnabled(false);
         }
     }
 
     @Override
-    public boolean onOptionsItemSelected(final MenuItem item) {
-        if (item.getItemId() == android.R.id.home) {
-            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
-            getSupportFragmentManager().popBackStack();
+    public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
+        mDrawer.closeDrawer(GravityCompat.START);
+
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (mDrawer.isDrawerOpen(GravityCompat.START)) {
+            mDrawer.closeDrawer(GravityCompat.START);
+        } else {
+            super.onBackPressed();
+        }
+    }
+
+    protected void initNavigationMenu() {
+        initMenuToggle();
+
+        final NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+        navigationView.setNavigationItemSelectedListener(this);
+    }
+
+    private void initMenuToggle() {
+        mDrawer = (DrawerLayout) findViewById(R.id.drawer_layout_main);
+
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
+        setSupportActionBar(toolbar);
+
+        mToggle = new ActionBarDrawerToggle(
+                this, mDrawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        mDrawer.addDrawerListener(mToggle);
+        mToggle.syncState();
+    }
+
+    private void isDrawerEnabled(final boolean isEnabled) {
```

Comment:
rethink method name. 
isDrawerEnabled is a good name for a method to return if enabled. Seems it's method to init or setup